### PR TITLE
DW-4424 and DW-4421

### PR DESCRIPTION
**Pull request by dan-hill2802**

### CODE REVIEW

**Comment by dan-hill2802** [comment]
> Weren't we going to have one of the date time fields to be DEFAULT CURRENT_TIMESTAMP - the write_timestamp perhaps?

this is now done

File: provisioner.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        "correlation_id": {
+            "Field": "correlation_id",
+            "Type": "varchar(45)",
+            "Null": "YES",
+            "Key": "",
+            "Default": None,
+        },
+        "topic_name": {
+            "Field": "topic_name",
+            "Type": "varchar(45)",
```
Comment (tea-short) [inline provisioner.py:96]: Again, 45 is probably too short

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    needs: get-publish-version
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+      - name: Zip lambda
+        run: mkdir artifacts
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt -t artifacts
+          cp main.py artifacts
```
[the remark this answers is not in the document]
Comment (dan-hill2802) [reply .github/workflows/main.yml:33]: yeah, cert didn't exist when I wrote this, good spot. I won't use that 2019-2015 cert as it was only put in to cover the transition back in February (only half of it is now valid)

[the remark this answers is not in the document]
Comment (tea-short) [reply .github/workflows/main.yml:33]: Yeah, the latest RDS CA cert is here:
https://s3.amazonaws.com/rds-downloads/rds-ca-2019-root.pem

It really depends if compatibility with any other RDS flavour (i.e. not Aurora Serverless) is required or not.

File: common.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    if "RDS_DATABASE_NAME" in os.environ:
+        _args["rds_database_name"] = os.environ["RDS_DATABASE_NAME"]
+
+    if "RDS_PASSWORD_SECRET_NAME" in os.environ:
+        _args["rds_password_secret_name"] = os.environ["RDS_PASSWORD_SECRET_NAME"]
+
+    # Validate event and environment variables
+    missing_event_keys = []
+    for required_message_key in required_keys:
```
Comment (tea-short) [inline common.py:72]: It appears that `required_keys` only contains one key, `table-name`, i.e. we aren't testing if things like `RDS_ENDPOINT` have been set - is this intentional?

Comment (dan-hill2802) [reply common.py:72]: yeah

File: .github/workflows/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dan-hill2802) [reply .github/workflows/main.yml:33]: also, the metadata store is never going to need both serverless and non-serverles, unlike the manage mysql users

File: common.py (hunk 3 above)
Comment (dan-hill2802) [reply common.py]: I've now added checking fro required env vars too https://github.com/dwp/dataworks-ingestion-metadata-interface/pull/1/commits/3bf11ef2474c1c724d9d5af8c2bdc74287025b0c

**Pull request merged**